eclass/blas.eclass

```diff
@@ -0,0 +1,241 @@
+# compatible with any BLAS version. Either it or BLAS_COMPAT must be set
+# before the `inherit' call. BLAS_COMPAT_ALL overrides BLAS_COMPAT.
+#
+# Example:
+# @CODE
+# BLAS_COMPAT_ALL=1
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_USEDEP
+# @DESCRIPTION:
+# This is an eclass-generated USE-dependency string which has to be used
+# to propagate the chosen BLAS-implementation down the dependency graph
+#
+# Example use:
+# @CODE
+# RDEPEND="sci-libs/foo[${BLAS_USEDEP}]"
+# @CODE
+#
+# Example value:
+# @CODE
+# refblas?,openblas?
+# @CODE
+
+
+# @ECLASS-VARIABLE: BLAS_CONDITIONAL_FLAG
+# @DESCRIPTION:
+# This variable contains the USE-flag that selects whether the package
+# should depend on BLAS or not.
+# If non-empty, it gets prepended to REQUIRED_USE, DEPEND and RDEPEND.
+# If it is an array, we will depend on BLAS when at least one of the 
+# USE-flags is activated (logical OR)
+#
+# If, for example, set to
+# @CODE
+# BLAS_CONDITIONAL_FLAG="blas"
+# @CODE
+#
+# Then the eclasses REQUIRED_USE and DEPENDs look like this:
+# @CODE
+# DEPEND="blas? ( blas_refblas? ( sci-libs/blas-reference ))"
+# RDEPEND="${DEPEND}"
+# REQUIRED_USE="blas? ( ^^ ( blas_refblas blas_openblas ) )"
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_USE_CBLAS
+# @DESCRIPTION:
+# This variable sets whether your package depends on BLAS C-headers.
+# If non-empty, additional packages will be added to the DEPEND and 
+# RDEPEND variable.
+# This variable must be set before the call to inherit
+#
+# Example usage:
+# @CODE
+# BLAS_USE_CBLAS=1
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_REQ_USE
+# @DESCRIPTION:
+# Adds a USE-flag requirement to every BLAS-implementation we depend on
+# (mainly useful for multilib, since the implementations share nearly no
+# other flags)
+# This must be set before the call to inherit blas
+#
+# Example usage:
+# @CODE
+# BLAS_REQ_USE="int64"
+# @CODE
+# 
+# Will result in DEPENDs like.
+# @CODE
+# DEPEND="blas_refblas? ( sci-libs/blas-reference[int64] )
+# @CODE
+
+
+
+BLAS_IMPLS=(refblas openblas gotoblas mkl)
+BLAS_SUPP_IMPLS=()
+
+_blas_provider_refblas="sci-libs/blas-reference"
+_blas_provider_openblas="sci-libs/openblas"
+_blas_provider_gotoblas="sci-libs/gotoblas2"
+_blas_provider_mkl="sci-libs/mkl"
+
+_cblas_provider_refblas="sci-libs/cblas-reference"
+_cblas_provider_openblas=""
+_cblas_provider_gotoblas=""
+_cblas_provider_mkl=""
+
+function _blas_impl_valid(){
+	local impl
+	for impl in "${BLAS_IMPLS[@]}"
+	do
+		if [ "$1" == "$impl" ]
+		then
+			return 0
+		fi
+	done
+	return 1
+}
+
+function _blas_best_impl(){
+	local impl
+	
+	for impl in "${BLAS_SUPP_IMPLS[@]}"
+	do
+		if use "$(_blas_useflag_by_impl $impl)"
+		then
+			echo $impl
+			return
+		fi
+	done
+	
+	die "No matching implementation found"
+}
+
+function _blas_useflag_by_impl(){
+	echo "${@/#/blas_}"
+}
+
+function _blas_name_by_impl(){
+	eval "echo \"\${_blas_name_$1}\""
+}
+
+function _blas_usedep(){
+	local tmp
+	tmp=("$(_blas_useflag_by_impl "${BLAS_SUPP_IMPLS[@]/%/?}" )" )
```

Comment:
The variable inside the array is quoted, so this array will always have one element. In other words, you can drop the outer `()` for the same effect.